src/platform/linux/mod.rs

```diff
@@ -25,18 +30,30 @@ pub enum UnixBackend {
     X(Arc<XConnection>),
     Wayland(Arc<wayland::WaylandContext>),
     Error(XNotSupported),
-} 
+}
 
 lazy_static!(
     pub static ref UNIX_BACKEND: UnixBackend = {
-        if let Some(ctxt) = wayland::WaylandContext::init() {
-            UnixBackend::Wayland(Arc::new(ctxt))
-        } else {
+        #[inline]
+        fn x_backend() -> UnixBackend {
             match XConnection::new(Some(x_error_callback)) {
                 Ok(x) => UnixBackend::X(Arc::new(x)),
                 Err(e) => UnixBackend::Error(e),
             }
         }
+        match env::var(BACKEND_PREFERENCE_ENV_VAR) {
+            Ok(ref s) if s == "x11" => {
+                println!("{}: using x11 backend", BACKEND_PREFERENCE_ENV_VAR);
```

Comment:
I'm not a fan of such unconditional printing on stdout. It'd be better on stderr imo.